clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -187,6 +205,18 @@ public RecordAppendResult append(TopicPartition tp,
                                      Header[] headers,
                                      Callback callback,
                                      long maxTimeToBlock) throws InterruptedException {
+        return append(tp, timestamp, key, value, headers, callback, maxTimeToBlock, null, false);
```

Comment:
This method is used only the tests. Can we remove it and redirect the usages to new method?

Reply:
I was just starting to do that yesterday. Glad we are on the same page!